### ISSUE TITLE
[Button] Accessibility fix for Primary, Secondary, and Danger buttons for dark themes

### ISSUE THREAD

**Issue by shixiedesign** [carbon-design-system/carbon#2729]
# Accessibility fix for Primary, Secondary, and Danger buttons on dark themes

**Problem** 

The 60s grade colors are not 3:1 accessible on G90 backgrounds, and we cannot use 50s grade colors for fill of the components because label text is white, and white is not 4.5:1 accessible on grade 50s colors. 

**Previous solution now ditched**
- add `1px inner border` to Primary, Secondary, and Danger buttons on Dark themes
- border colors are in spec image below

A few more design issues to resolve here before it is ready for dev. This issue is currently blocked by 
- adding $inverse-suppport tokens https://github.com/carbon-design-system/carbon/issues/2684
- secondary button's border color has no token. TBD 

![Button dark themes2](https://user-images.githubusercontent.com/15144993/57789262-29689000-76fe-11e9-8a36-55765c5ce6bd.png)

**Comment by elizabethsjudd** [comment]
For my own curiosity can you explain to me how adding a border makes this accessible?  How was the white text on the colored buttons not accessible before?

**Comment by shixiedesign** [comment]
Hi @elizabethsjudd, I see what you are asking and yes this has been a bit of a debate on our side too. The underline makes sure component has a visual indicator that's 3:1 contrast against page background, Gray90 & Gray100. We fixed this with an underline because we cannot lighten the button fill color any more, because white text has to be 4.5:1 accessible against the button fill. So this is a fix to satisfy the rule, and arguably without real improvement to the component.

**Comment by shixiedesign** [comment]
We are likely gonna change from underline to a full border, which would make more sense, and would actually help with accessibility. Waiting on getting in alignment. 

Edit: YES WE ARE GOING FULL BORDER! Updated spec image above.

**Comment by shixiedesign** [comment]
### Alternative solution

Ultimate goal is to make sure buttons are accessible against background 3:1 and button labels are accessible against button fill 4.5:1. 

An alternative solution is to adjust our color palette. Adding border would cause issues for light theme buttons. Below buttons uses a lightened up version of the 60s grade colors:

![Button dark themes Copy](https://user-images.githubusercontent.com/15144993/57905082-e9f38e00-783a-11e9-8c2d-d84b46e517cd.png)

**Comment by shixiedesign** [comment]
## Shixie's tweaked palette WIP

Continuing to pursue the alternative solution - tweak the palette. Presented to Sadek. Feedback: 
- [x] Test gray 10 on poor monitors to make sure it is not too light and going lost

![image](https://user-images.githubusercontent.com/15144993/58303283-0b152b00-7db5-11e9-987d-589d717a3239.png)
![image](https://user-images.githubusercontent.com/15144993/58303311-297b2680-7db5-11e9-9586-0620d7fdd909.png)
![image](https://user-images.githubusercontent.com/15144993/58303317-2c761700-7db5-11e9-8c2e-7042f539aaba.png)

**Comment by shixiedesign** [comment]
## Update
This issue no longer describes the work done to address it. Making a new issue for palette changes, and making it a dependency of this issue. Once new palette is merged, button accessibility issue should be automatically fixed via tokens getting new values. 

Palette v2 issue https://github.com/carbon-design-system/carbon/issues/3226
